receiver/googlecloudspannerreceiver/internal/filterfactory/testhelpers_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filterfactory
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/model/pdata"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/filter"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudspannerreceiver/internal/metadata"
+)
+
+const (
+	metricFullName = "metricFullName"
+	prefix1        = "prefix1-"
+	prefix2        = "prefix2-"
+)
+
+type errorOnShutdownFilter struct {
```

Comment:
You may be better off using something like https://pkg.go.dev/github.com/stretchr/testify/mock which may mocking scenarios like this slightly easier.

Reply:
No worries :) I had used it inside the `awskinesisexporter` if you want an example?

Reply:
@MovieStoreGuy , I've just committed mock related changes. In this particular case I do not see a lot of advantages, but anyway - it is something new to learn.